src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -1912,16 +1910,18 @@ private FunctionGroup FirstFunctionGroupInInheritanceChain(RuntimeCore runtimeCo
             if (ret.IsNull)
             {
                 //wipe the trace cache
-                TraceUtils.ClearTLSKey(TRACE_KEY);
+                TraceUtils.ClearAllKnownTLSKeys();
+                newTraceData.Data = null;
             }
-
```

Comment:
The cases where ret is not null, it checks for the trace data in the local storage. This happens on every execution of a node.